test/e2e/mco_test.go

```diff
@@ -132,3 +136,72 @@ func TestClusterOperatorStatusExtension(t *testing.T) {
 	require.True(t, ok)
 	require.Equal(t, "extension", v)
 }
+
+func TestMetrics(t *testing.T) {
+	cs := framework.NewClientSet("")
+
+	delete := helpers.CreateMCP(t, cs, "infra")
+	// Create infra pool to roll out MC changes
+	workerOldMc := helpers.GetMcName(t, cs, "worker")
+	unlabelFunc := helpers.LabelRandomNodeFromPool(t, cs, "worker", "node-role.kubernetes.io/infra")
+	oldInfraConfig := helpers.CreateMC("old-infra", "infra")
+
+	t.Cleanup(func() {
+		node := helpers.GetSingleNodeByRole(t, cs, "infra")
+		helpers.ExecCmdOnNode(t, cs, node, "/bin/bash", "-c", "sed -e s/wrong-data-here//g -i /rootfs/etc/containers/storage.conf")
+		unlabelFunc()
+		// wait for the mcp to go back to previous config
+		if err := helpers.WaitForPoolComplete(t, cs, "worker", workerOldMc); err != nil {
+			t.Fatal(err)
+		}
+		delete()
+		require.Nil(t, cs.MachineConfigs().Delete(context.TODO(), oldInfraConfig.Name, metav1.DeleteOptions{}))
+
+	})
+
+	_, err := cs.ClusterOperators().Get(context.TODO(), "machine-config", metav1.GetOptions{})
+	require.Nil(t, err)
+
+	// create old mc to have something to verify we successfully rolled back
+	_, err = cs.MachineConfigs().Create(context.TODO(), oldInfraConfig, metav1.CreateOptions{})
+	require.Nil(t, err)
+	oldInfraRenderedConfig, err := helpers.WaitForRenderedConfig(t, cs, "infra", oldInfraConfig.Name)
+	err = helpers.WaitForPoolComplete(t, cs, "infra", oldInfraRenderedConfig)
+	require.Nil(t, err)
+
+	node := helpers.GetSingleNodeByRole(t, cs, "infra")
+
+	// Get the machine config pool
+	mcp, err := cs.MachineConfigPools().Get(context.TODO(), "infra", metav1.GetOptions{})
+	require.Nil(t, err)
+
+	e2e_shared_test.MutateNodeAndWait(t, cs, &node, mcp)
```

Comment:
in my cleanup, I do the following: `helpers.ExecCmdOnNode(t, cs, node, "/bin/bash", "-c", "sed -e s/wrong-data-here//g -i /rootfs/etc/containers/storage.conf")` to un-degrade the node since it removes the data done in `MutateNodeAndWait`

Reply:
Aha! I missed that. Thanks for pointing it out!